### PR TITLE
REQ-627 CA-334557 de-quarantine Intel GPUs

### PR DESCRIPTION
**Pull request by lindig**
* widen de-quarantining to include Intel GPUs (GVT_g)
* limit de-quarantining to GPUs where this is actually required
* make policy for de-quarantining more explicit

These commit are not rebased onto the current master branch because re-basing introduces conflicts but these commits can be merged without conflict.

### CODE REVIEW

File: lib/xenops_server.ml [hunk 1]
```diff
@@ -1919,12 +1933,13 @@ and perform_exn ?subtask ?result (op: operation) (t: Xenops_task.task_handle) :
...
           List.concat
-            [ List.map (fun pci -> PCI_plug(pci.Pci.id, false)) pcis_sriov
+            [ dequarantine_ops vgpus' pcis
+            ; List.map (fun pci -> PCI_plug(pci.Pci.id, false)) pcis_sriov
             ; [VGPU_start (vgpus, true)]
             ]
         in
         perform_atomics (List.concat
-          [ List.map (fun pci -> PCI_dequarantine pci.Pci.id) pcis
+          [ List.map (fun pci -> PCI_dequarantine pci.Pci.address) pcis
```
Comment (robhoes) [inline lib/xenops_server.ml]: Isn't this covered by the `dequarantine_ops` above?

File: lib/xenops_server.ml [hunk 2]
```diff
@@ -932,6 +932,23 @@ let is_nvidia_sriov vgpus pci =
...
+  | Nvidia _ -> true
+  | _ -> false
+  )
+
+(* compute a list of PCI_dequarantine operations for PCI devices *)
+let dequarantine_ops vgpus pcis =
+  List.concat
+  [ vgpus |> List.filter must_dequarantine |> List.map
+        Xenops_interface.Vgpu.(fun vgpu -> PCI_dequarantine vgpu.physical_pci_address)
+  ; pcis  |> List.filter (is_nvidia_sriov vgpus) |> List.map
```
Comment (robhoes) [inline lib/xenops_server.ml]: And MXGPU?

File: lib/xenops_server.ml (hunk 1 above)
Comment (lindig) [reply lib/xenops_server.ml]: Yes. This line needs to go.

File: lib/xenops_server.ml (hunk 2 above)
Comment (lindig) [reply lib/xenops_server.ml]: As far as I understand it does not need to be de-quarantined. Am i wrong?

**Comment by lindig** [comment]
I have further simplified the fix and re-tested it. I am squashing commits and would suggest to review ait again.

**Comment by lindig** [comment]
I've now rebased because this could no longer be merged.

**Pull request merged**